core/src/main/java/com/datastax/oss/driver/internal/core/metadata/LoadBalancingPolicyWrapper.java

```diff
@@ -198,11 +195,13 @@ private void processNodeStateEvent(NodeStateEvent event) {
     }
   }
 
-  private static Map<InetSocketAddress, Node> excludeDownHosts(Metadata metadata) {
-    ImmutableMap.Builder<InetSocketAddress, Node> nodes = ImmutableMap.builder();
-    for (Node node : metadata.getNodes().values()) {
+  private static Map<UUID, Node> excludeDownHosts(Metadata metadata) {
+    ImmutableMap.Builder<UUID, Node> nodes = ImmutableMap.builder();
+    for (Map.Entry<UUID, Node> entry : metadata.getNodes().entrySet()) {
+      UUID id = entry.getKey();
+      Node node = entry.getValue();
       if (node.getState() == NodeState.UP || node.getState() == NodeState.UNKNOWN) {
```

Comment:
why are we treating `UKNOWN` as not down?

Reply:
`UNKNOWN` only indicates that the driver has not tried to connect yet, the LBP will still want to assign a distance to those nodes.

Actually while reading this I'm now questioning the relevance of even excluding `DOWN` nodes. I'll create another ticket.